config/schemas/6/rapidast_schema.json

```diff
@@ -0,0 +1,605 @@
+                        "envFile": {
+                            "type": "string"
+                        }
+                    },
+                    "required": []
+                },
+                "googleCloudStorage": {
+                    "type": "object",
+                    "properties": {
+                        "keyFile": {
+                            "type": "string"
+                        },
+                        "bucketName": {
+                            "default": "default-bucket-name",
+                            "type": "string"
+                        },
+                        "directory": {
+                            "type": "string"
+                        }
+                    },
+                    "comment": "Credentials are optional, as the client defaults to those inferred from the environment",
+                    "required": []
+                },
+                "defectDojo": {
+                    "type": "object",
+                    "properties": {
+                        "url": {
+                            "type": "string"
+                        },
+                        "authorization": {
+                            "type": "object",
+                            "properties": {
+                                "username": {
+                                    "type": "string"
+                                },
+                                "password": {
+                                    "type": "string"
+                                },
+                                "token": {
+                                    "type": "string"
+                                }
+                            }
+                        },
+                        "ssl": {
+                            "default": true,
+                            "type": "boolean"
+                        }
+                    },
+                    "oneOf": [
+                        {
+                            "required": [
+                                "url",
+                                "username",
+                                "password"
+                            ]
+                        },
+                        {
+                            "required": [
+                                "url",
+                                "token"
+                            ]
+                        }
+                    ]
+                }
+            },
+            "required": []
+        },
+        "application": {
+            "type": "object",
+            "properties": {
+                "shortName": {
+                    "default": "scannedApp",
+                    "type": "string"
+                },
+                "ProductName": {
+                    "type": "string"
+                },
+                "url": {
+                    "type": "string",
+                    "format": "uri"
+                }
+            },
+            "required": []
```

Comment:
The `url` should be required in certain cases, such as when a 'zap' scanner is enabled. However, implementing this logic with the current model is a bit complex, identifying the active scanners is not a trivial task. We can revisit and improve this in the future

Reply:
ok to revisit later. Ideally the option should be moved to each scanner that requires it.